library/network_connections.py

```diff
@@ -222,31 +222,29 @@ def link_infos(cls, refresh=False):
         return linkinfos
 
     @classmethod
-    def link_info_find(cls, refresh=False, mac=None, ifname=None):
-        if mac is not None:
+    def link_info_find(cls, mac=None, ifname=None):
+        if mac:
             mac = Util.mac_norm(mac)
-        for linkinfo in cls.link_infos(refresh).values():
-            perm_address = linkinfo.get("perm-address", None)
-            current_address = linkinfo.get("address", None)
 
-            # Match by perm-address (prioritized)
-            if mac is not None and perm_address not in [None, "00:00:00:00:00:00"]:
-                if mac == perm_address:
-                    return linkinfo
+        result = None
+        null_mac = "00:00:00:00:00:00"
 
-            # Fallback to match by address
-            if mac is not None and (perm_address in [None, "00:00:00:00:00:00"]):
-                if mac == current_address:
-                    matched_by_address = linkinfo  # Save for potential fallback
+        for linkinfo in cls.link_infos().values():
+            perm_address = linkinfo.get("perm-address", null_mac)
+            current_address = linkinfo.get("address", null_mac)
 
-            if ifname is not None and ifname == linkinfo.get("ifname", None):
-                return linkinfo
+            if ifname and ifname == linkinfo["ifname"]:
+                result = linkinfo
+                break
 
-        # Return fallback match by address if no perm-address match found
-        if "matched_by_address" in locals():
-            return matched_by_address
+            if mac:
+                if perm_address != null_mac and mac == perm_address:
```

Comment:
What's the reason to check that the permanent mac address is not null here?

Reply:
null mac is an indication of the absence of the mac address or uninitialized mac address,  checking the permanent mac address is not null here can guarantee that the permanent mac address is valid and initialized.
As alluded in https://standards-support.ieee.org/hc/en-us/articles/4888705676564-Guidelines-for-Use-of-Extended-Unique-Identifier-EUI-Organizationally-Unique-Identifier-OUI-and-Company-ID-CID,  `EUI-48 and EUI-64 identifiers ( Extended Unique Identifier) are most commonly used as globally unique network addresses (sometimes called MAC addresses)`,  and `00-00-00-00-00-00 shall not be used as identifiers`.

Reply:
The function `get_perm_addr()` effectively converts the unset/uninitialized MAC into `00:00:00:00:00:00` 
https://github.com/linux-system-roles/network/blob/main/module_utils/network_lsr/ethtool.py#L39.